src/trading_rl_agent/monitoring/metrics.py

```
@@ -0,0 +1,27 @@
+
+
+class MetricsCollector:
+    """Collects and logs metrics during trading and backtesting."""
+
+    def __init__(self) -> None:
+        self.logger = get_logger(self.__class__.__name__)
+        self.history: List[Dict[str, Any]] = []
+
+    def log_metrics(self, metrics: Any) -> None:
+        """Log metrics and store them in memory."""
+        metrics_dict = convert_metrics_to_dict(metrics)
```

Ruff (0.11.9)

Undefined name `convert_metrics_to_dict`

(F821)